source/mbed_trace.c

```diff
@@ -259,6 +259,10 @@ static void mbed_trace_default_print(const char *str)
 }
 void mbed_tracef(uint8_t dlevel, const char *grp, const char *fmt, ...)
 {
+    if (NULL == m_trace.line) {
+        return;
```

Comment:
there should be also tmp resetting, like R266/270: `mbed_trace_reset_tmp();`
Perhaps better if you little reorder lines 262-272 so that there is no duplicate code so much, like
```
if (mbed_trace_skip(dlevel, grp) || fmt == 0 || grp == 0 || !m_trace.printf || !m_trace.line) {
     //return tmp data pointer back to the beginning
     mbed_trace_reset_tmp();
     return;
}
m_trace.line[0] = 0;
```